src/core/meshcop/tcat_agent.cpp

```diff
@@ -481,6 +483,26 @@ Error TcatAgent::HandleSetActiveOperationalDataset(const Message &aIncommingMess
     return error;
 }
 
+Error TcatAgent::HandleDecomission(void)
+{
+    Error error = kErrorNone;
+
+    IgnoreReturnValue(otThreadSetEnabled(&GetInstance(), false));
+    Get<ActiveDatasetManager>().Clear();
+    Get<PendingDatasetManager>().Clear();
+    Get<Settings>().Wipe();
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    Get<KeyManager>().DestroyTemporaryKeys();
+    Get<KeyManager>().DestroyPersistentKeys();
+#else
+    NetworkKey networkKey;
+    networkKey.Clear();
+    Get<KeyManager>().SetNetworkKey(networkKey);
+#endif
```

Comment:
@abtink Can You take a look at this function? As @maciejbaczmanski suggested we could move this implementation to `ErasePersistentInfo`  and call it here since this implementation looks redundant.

Reply:
I agree that using `ErasePersistentInfo()` here is the right approach. I believe it already removes key references as well.

Regarding adding `Get<KeyManager>().SetNetworkKey(networkKey)` with a cleared key within `ErasePersistentInfo()`, I'm not sure if it a good way. While clearing the key from RAM has benefits, it doesn't align with the purpose of `ErasePersistentInfo()`, which should only erase persisted information, not data in RAM.

For clearing RAM, we have `Instance::Reset()` (mapped to `otPlatReset()`), which aims to clear all information in RAM (at minimum, it should invoke the `Instance` constructor again). However, it's often implemented as a full device reset.

Reply:
Resetting the device in this case is not an option for us since the Thread speck clearly states that after the decommissioning it should be possible to get a Commissioner certificate in the same session. In order to do this we need to store the cert in RAM. (This will be a part of next PR). So how about:
```

Error TcatAgent::HandleDecomission(void)
{
    Error error = kErrorNone;

    IgnoreReturnValue(otThreadSetEnabled(&GetInstance(), false));
    Get<ActiveDatasetManager>().Clear();
    Get<PendingDatasetManager>().Clear();

    error = Get<Instance>.ErasePersistentInfo();

#if !OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
    NetworkKey networkKey;
    networkKey.Clear();
    Get<KeyManager>().SetNetworkKey(networkKey);
#endif

    return error;
}
```

Reply:
Right, the intention after a "Decommission" command is that the device doesn't reboot and keeps the session/link with the TCAT Commissioner open for further command processing. (It could be a "Commission" command again in the same session.)